File: Maths_And_Stats/Geometry/N_Dimensional_Distance/n_dimensional_distance.rb
#
# Calculate N-dimensional distance
#
# @param [Array] vector_a
# @param [Array] vector_b
#
# @return [Float] distance
#
def n_dims_distance(vector_a, vector_b)
  raise "Dimension mismatch" unless vector_a.length == vector_b.length
  sum = vector_a.zip(vector_b)
    .map { |a, b| (a - b)**2 }
    .sum
  Math.sqrt(sum)
end

puts "Enter vector value separated by comma, like 1,2,3,4,5"
print "First vector => "
vector_a = gets.chomp.split(",").map(&:to_f)
print "Second vector => "
vector_b = gets.chomp.split(",").map(&:to_f)
puts "Distance: #{n_dims_distance(vector_a, vector_b).round(3)}"

File: Maths_And_Stats/Geometry/N_Dimensional_Distance/N_Dimensional_Distance.c
#include <stdio.h>
#include <stdlib.h>
#include <math.h>

/**
 * This function calculates the euclidian (or L2) distance of two points.
 * @param point1 coordinates of first point stored in an array
 * @param point2 coordinates of second point stored in an array
 * @param dimension the number of coordinates each point is expected to have
 *
 * @return The euclidian distance between the points
 */
float n_dimensional_distance(double const *point1, double const * point2, const size_t dimension)
{
	double result = 0;

	for (size_t index = 0; index < dimension; ++index)
	{
		double difference = point1[index] - point2[index];
		result += difference*difference;
	}

	return sqrt(result);
}

int main(void)
{
	const size_t N = 3;
	const double point1[] = {1,1,1};
	const double point2[] = {3,4,7};

	double distance = n_dimensional_distance(point1, point2, N);

	printf("The %zu-dimensional distance between the points point1 and point2 is %f", N, distance);

	return EXIT_SUCCESS;
}
